pkg/analyzer/analyzer.go

```diff
@@ -35,12 +35,14 @@ type analyzer struct {
 // The "-p" argument can be used to add a pattern.
 // Set "-examples" to analyze godoc examples
 // Set "-permit=false" to ignore "//permit:<identifier>" directives.
-func NewAnalyzer() *analysis.Analyzer {
+func NewAnalyzer(patterns ...string) *analysis.Analyzer {
```

Comment:
Is this change just for testing?  It seems like an exception from the rest of the controls (and how I think analyzers are initialized without arguments in general, so is it absolutely necessary?

Reply:
Yes.

It's not necessary, I replaced it with:
```
	a := analyzer.NewAnalyzer()
	for _, pattern := range patterns {
		if err := a.Flags.Set("p", pattern); err != nil {
			t.Fatalf("unexpected error when setting pattern: %v", err)
		}
	}
```